src/js/view/view-model.ts

```diff
@@ -1,8 +1,9 @@
 import SimpleModel from 'model/simplemodel';
 import type Model from 'controller/model';
 import type { MediaModel, PlayerModelAttributes, MediaModelAttributes } from 'controller/model';
-import type { DefaultProvider, GenericObject } from 'types/generic.type';
+import type { GenericObject } from 'types/generic.type';
 import type AdProgramController from 'program/ad-program-controller';
+import type { ProviderWithMixins } from 'providers/default';
```

Comment:
Use a more generic type for provider like `ImplementedProvider` here.

Reply:
We should strive to remove public access to the provider from models, but this is how they are used (if you search for `getVideo` throughout the projects:

- write to `.src` to reset hls.js state in ad-program-controller
- read-only access to `.setFullscreen` in view.js
- read-only access to `.resize` in view.js
- read-only access to `.getFullScreen` in view.js
- read-only access to `.setContainer` in view.js

The only odd one out is `src`. If we add that to `ImplementedProvider` for now that would be best since a public property in an interface (with a comment) would signal that it is nulled externally and not required by all providers.

Reply:
:+1: replacing